### PR TITLE
BUG: signal: Fix error raised by savgol_coeffs when deriv > polyorder.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
An example of the problem:

    >>> from scipy.signal import savgol_coeffs
    >>> savgol_coeffs(window_length=7, polyorder=3, deriv=4)
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "[...]/site-packages/scipy/signal/_savitzky_golay.py", line 132, in savgol_coeffs
        y[deriv] = factorial(deriv) / (delta ** deriv)
    IndexError: index 4 is out of bounds for axis 0 with size 4

The correct result in this case is an array of zeros: for a
polynomial of degree m, the derivative of order n > m is
identically 0.  After this change, that's what we get:

    >>> from scipy.signal import savgol_coeffs
    >>> savgol_coeffs(window_length=7, polyorder=3, deriv=4)
    array([0., 0., 0., 0., 0., 0., 0.])

I also updated the outputs shown in the docstrings of savgol_coeffs
and savgol_filter.

### CODE REVIEW

**Comment by WarrenWeckesser** [comment]
This is a pretty simple fix, so I've added it to the 1.3 milestone.

File: scipy/signal/_savitzky_golay.py [hunk 1]
```diff
@@ -62,8 +62,8 @@ def savgol_coeffs(window_length, polyorder, deriv=0, delta=1.0, pos=None,
     >>> savgol_coeffs(5, 2)
     array([-0.08571429,  0.34285714,  0.48571429,  0.34285714, -0.08571429])
     >>> savgol_coeffs(5, 2, deriv=1)
-    array([  2.00000000e-01,   1.00000000e-01,   2.00607895e-16,
-            -1.00000000e-01,  -2.00000000e-01])
+    array([ 2.00000000e-01,  1.00000000e-01,  2.07548111e-16, -1.00000000e-01,
+           -2.00000000e-01])
```
Comment (tylerjereddy) [inline scipy/signal/_savitzky_golay.py]: Scanning over `refguide_check.py`, my initial guess is that this isn't opted-in for testing, especially given that this is a "private module." That could be a matter for another PR though. If it is worth updating the docstring for new results, it is perhaps worth having it tested.

File: scipy/signal/tests/test_savitzky_golay.py [hunk 2]
```diff
@@ -137,6 +137,19 @@ def test_sg_coeffs_deriv():
         assert_allclose(coeffs2.dot(x), d2x[pos], atol=1e-10)
 
 
+def test_sg_deriv_ge_polyorder():
+    """
+    If deriv >= polyorder, the coefficients should be all 0.
```
Comment (tylerjereddy) [inline scipy/signal/tests/test_savitzky_golay.py]: Why does the docstring for the unit test claim the condition for all zero is `>=`, while the code change above uses `>`? Maybe one needs a slight adjustment?

I'm guessing you need the derivative order to be `>` so that you end up taking the derivative of a constant. Perhaps with `==` order you can end up with a non-zero constant like standard calculus?

Comment (WarrenWeckesser) [reply scipy/signal/tests/test_savitzky_golay.py]: > Why does the docstring for the unit test claim the condition for all zero is `>=`, while the code change above uses `>`?

Because that docstring in the test is wrong. 😧   I'll fix it.

File: scipy/signal/_savitzky_golay.py (hunk 1 above)
Comment (WarrenWeckesser) [reply scipy/signal/_savitzky_golay.py]: I haven't been following the developments in the doc-building tools very closely.  The names `savgol_coeffs` and `savgol_filter` end up in the `signal` namespace.  Are they not checked like everything else?

Comment (tylerjereddy) [reply scipy/signal/_savitzky_golay.py]: Ah, maybe they are then--I can always check locally by breaking the docstring intentionally in case these deviations weren't sufficient, and then running refguide.

File: scipy/signal/tests/test_savitzky_golay.py (hunk 2 above)
Comment (WarrenWeckesser) [reply scipy/signal/tests/test_savitzky_golay.py]: I pushed an update with the test's docstring fixed.

File: scipy/signal/_savitzky_golay.py (hunk 1 above)
Comment (WarrenWeckesser) [reply scipy/signal/_savitzky_golay.py]: > I can always check locally by breaking the docstring intentionally in case these deviations weren't sufficient, and then running refguide.

OK, I just tried that.  In the `savgol_coeffs` docstring, I locally changed the last output value from 6.0 to 16.0.  When I ran `python refguide_check.py --doctests --verbose signal`, the example failed:

```
Running checks for 1 modules:
scipy.signal ..........................................................................................................................................................................................................................................................................................................F...................................

============
scipy.signal
============

Non-deprecated objects in __all__: 176
Objects in refguide: 156

scipy.signal.savgol_coeffs
--------------------------

File "/Users/warren/miniconda36/lib/python3.6/site-packages/scipy-1.3.0.dev0_-py3.6-macosx-10.7-x86_64.egg/scipy/signal/_savitzky_golay.py", line 82, in savgol_coeffs
Failed example:
    c.dot(x)
Expected:
    16.0
Got:
    6.0


ERROR: refguide or doctests have errors
```

When the docstring has the correct value, the script runs with no errors.

**Review by larsoner — APPROVED** [review]
+1 for merge from me

I assume that the refguide stuff has to do with the refguide check having relative/absolute tolerance levels for the outputs that make the changes here considered to be within the acceptable numerical noise levels.

**Comment by tylerjereddy** [comment]
Great, thanks @WarrenWeckesser

**Comment by WarrenWeckesser** [comment]
Thanks Tyler & Eric.
